website/status.php

```diff
@@ -1,37 +1,57 @@
 <?php
+
 @define('CONST_ConnectionBucket_PageType', 'Status');
 
 require_once(dirname(dirname(__FILE__)).'/settings/settings.php');
 require_once(CONST_BasePath.'/lib/init-website.php');
+require_once(CONST_BasePath.'/lib/ParameterParser.php');
+require_once(CONST_BasePath.'/lib/Status.php');
 
-
-function statusError($sMsg)
-{
-    header('HTTP/1.0 500 Internal Server Error');
-    echo 'ERROR: '.$sMsg;
-    exit;
-}
+$oParams = new Nominatim\ParameterParser();
+$sOutputFormat = $oParams->getSet('format', array('text', 'json'), 'text');
 
 $oDB =& DB::connect(CONST_Database_DSN, false);
-if (!$oDB || PEAR::isError($oDB)) {
-    statusError('No database');
-}
+$oStatus = new Nominatim\Status($oDB);
+
+$aResponse = [
+              'status' => 'ok',
+              'code' => 200,
+              'message' => 'OK'
+             ];
+
+$sErrorMsg = $oStatus->status(); // can be nil
```

Comment:
Sorry but I'm still not happy with that. Using the null value like that is confusing.

Let's go back to the format of the response message. Why does the code field need to be a HTTP error at all? The HTTP error is already sent with the HTTP header. How about a generic response that contains a status field as integer with 0 status ok and a Nominatim-specific error code otherwise and the message string.